### PR TITLE
Several refactorings

### PR DESCRIPTION
**Pull request by ChrisBr**
This PR adds some more specs around exceptions and refactors some of the code. The commits should be self explanatory.

### CODE REVIEW

**Comment by ChrisBr** [comment]
~dd37365 will get fixed with https://github.com/influxdata/influxdb-rails/pull/101 and can get dropped after it is merged.~

File: spec/support/broken_client.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require File.expand_path(File.dirname(__FILE__) + "/test_client")
+require "launchy"
```
Comment (hennevogel) [inline spec/support/broken_client.rb]: launchy?

File: lib/influxdb/rails/railtie.rb [hunk 2]
```diff
@@ -32,12 +33,18 @@ class Railtie < ::Rails::Railtie # :nodoc:
         }.each do |hook_name, subscriber_class|
           subscribe_to(hook_name, subscriber_class)
         end
+
+        reset_cache = lambda do |_, _, _, _, _|
+          InfluxDB::Rails.current.reset
+        end
+        ActiveSupport::Notifications.subscribe "process_action.action_controller", &reset_cache
```
Comment (hennevogel) [inline lib/influxdb/rails/railtie.rb]: I find this less clear. What's the reason to move it here?

**Review by hennevogel — CHANGES_REQUESTED** [review]

File: lib/influxdb/rails/railtie.rb (hunk 2 above)
Comment (ChrisBr) [reply lib/influxdb/rails/railtie.rb]: Don't remember why I moved it to the Railties so I moved it back.

**Review by hennevogel — APPROVED** [review]

**Pull request merged**